app/src/androidTest/java/ch/epfl/polybazaar/FavoritesTest.java

```diff
@@ -0,0 +1,61 @@
+package ch.epfl.polybazaar;
+import android.view.View;
+
+import androidx.test.rule.ActivityTestRule;
+
+import org.hamcrest.Matcher;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import ch.epfl.polybazaar.UI.SalesOverview;
+import ch.epfl.polybazaar.login.AuthenticatorFactory;
+import ch.epfl.polybazaar.login.MockAuthenticator;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.RootMatchers.withDecorView;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static ch.epfl.polybazaar.database.datastore.DataStoreFactory.useMockDataStore;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.core.IsNot.not;
+
+
+
+public class FavoritesTest {
+
+
+    MockAuthenticator auth;
+
+    @Rule
+    public final ActivityTestRule<SalesOverview> mActivityRule =
+            new ActivityTestRule<SalesOverview>(SalesOverview.class){
+            };
+
+
+    @Before
+    public void init() {
+        useMockDataStore();
+        auth = MockAuthenticator.getInstance();
+        AuthenticatorFactory.setDependency(auth);
+    }
```

Comment:
Shouldn't this code be written into a @BeforeClass ? The @Before will execute only before the following test and it might cause problems if other tests using the authenticator are created in the same file

Reply:
The code marked @Before is executed before each test, while @BeforeClass runs once before the entire test fixture. So I think here @Before is fine